dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/ComponentPredicate.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.tracking
+
+/**
+ * Used to decide which Android component of your application should be considered
+ * as a RUM View Event instance in a [ViewTrackingStrategy].
+ * @see FragmentViewTrackingStrategy
+ * @see ActivityViewTrackingStrategy
+ */
+interface ComponentPredicate<T> {
+    /**
+     * Decides whether the provided component should be considered as a RUM View.
+     * Make sure you are consistent when you validate a component otherwise you may experience
+     * some weird behaviours (e.g. by returning true in a first interrogation for a component
+     * followed by false in a second interrogation we might alter the data in the in progress
+     * View event or we might close the current valid View prematurely).
+     * @param component a component whose state changed
+     * @return true if you want the component to be tracked as a View, false otherwise
+     *
+     */
+    fun accept(component: T): Boolean
+
+    /**
+     * Executes the provided operation if the predicate verifies the argument.
+     * @param arg to be verified
+     * @param operation to be executed
+     */
+    fun runIfValid(arg: T, operation: (T) -> Unit) {
+        if (accept(arg)) {
+            operation(arg)
+        }
+    }
```

Comment:
I think this should be an extension function. Here you give a default implementation, but customers could override it if they want and change the way the library works.